### PR TITLE
docker_container: remove timeout when waiting for container to finish

### LINKED ISSUE #526: Docker container command throws timeout error
##### SUMMARY
Long-running docker commands seem to throw a "read timed out" error despite them processing. This did not happen with previous versions.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
docker_container

##### ANSIBLE VERSION
```paste below
ansible [core 2.14.0]
  config file = /home/user/.ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.11.0 (main, Oct 24 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-2)] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION
```paste below
Collection       Version
---------------- -------
community.docker 3.2.1
```

##### CONFIGURATION
```paste below
CONFIG_FILE() = /home/user/.ansible.cfg
```

##### OS / ENVIRONMENT
Fedora 37

##### STEPS TO REPRODUCE

1. `sudo dnf install podman-docker && sudo systemctl start podman`
2. Run `ansible-playbook -K test-docker-command.yml`
3. The command will throw an error, despite it still running fine (`sudo docker logs -f osm-import`).

```yaml
- name: Deploy
  hosts: localhost
  become: yes
  tasks:
    - name: Import data
      docker_container:
        name: osm-import
        command: import
        detach: no
        cleanup: yes
        image: overv/openstreetmap-tile-server
```

##### EXPECTED RESULTS
Completes successfully.

##### ACTUAL RESULTS

```paste below
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: requests.exceptions.ConnectionError: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out.
fatal: [localhost]: FAILED! => {"changed": false, "msg": "An unexpected requests error occurred when trying to talk to the Docker daemon: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out."}
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #526.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
docker_container

### CODE REVIEW

**Comment by felixfontein** [comment]
I don't really want to add a test since this will add > 1 minute runtime to CI (due to the nature of the bug).

File: plugins/module_utils/module_container/docker_api.py [hunk 1]
```diff
@@ -269,8 +269,8 @@ def create_container(self, client, container_name, create_parameters):
     def start_container(self, client, container_id):
         client.post_json('/containers/{0}/start', container_id)
 
-    def wait_for_container(self, client, container_id):
-        return client.post_json_to_json('/containers/{0}/wait', container_id)['StatusCode']
+    def wait_for_container(self, client, container_id, timeout=None):
+        return client.post_json_to_json('/containers/{0}/wait', container_id, timeout=timeout)['StatusCode']
```
Comment (felixfontein) [inline plugins/module_utils/module_container/docker_api.py:273]: passing `timeout=None` here basically disables timeout.

Comment (felixfontein) [reply plugins/module_utils/module_container/docker_api.py:273]: (This is also what Docker SDK for Python does: https://github.com/docker/docker-py/blob/main/docker/api/container.py#L1288-L1320)

**Review by kristianheljas — APPROVED** [review]
Hunky-dory!

**Comment by felixfontein** [comment]
@kristianheljas thanks for reviewing this!

**Pull request merged**